stress/shell/src/main/java/alluxio/stress/cli/MaxFileBench.java

```diff
@@ -0,0 +1,224 @@
+
+    JCommander jc = new JCommander(this);
+    jc.setAllowParameterOverwriting(true);
+    jc.setProgramName(this.getClass().getSimpleName());
+    try {
+      jc.parse(argsList.toArray(new String[0]));
+      if (mBaseParameters.mHelp) {
+        System.out.println(getBenchDescription());
+        jc.usage();
+        System.exit(0);
+      }
+    } catch (Exception e) {
+      LOG.error("Failed to parse command: ", e);
+      System.out.println(getBenchDescription());
+      jc.usage();
+      throw e;
+    }
+  }
+
+  @Override
+  public MasterBenchTaskResult runLocal() throws Exception {
+    ExecutorService service = ExecutorServiceFactories.fixedThreadPool("maxfile-bench-thread",
+        mParameters.mThreads).create();
+
+    List<Callable<Void>> callables = new ArrayList<>(mParameters.mThreads);
+    for (int i = 0; i < mParameters.mThreads; i++) {
+      callables.add(new AlluxioNativeMaxFileThread(i, mCachedNativeFs[i % mCachedNativeFs.length]));
+    }
+    LOG.info("Starting {} bench threads", callables.size());
+    long startMs = CommonUtils.getCurrentMs();
+    service.invokeAll(callables, FormatUtils.parseTimeSize(mBaseParameters.mBenchTimeout),
+        TimeUnit.MILLISECONDS);
+    mTotalResults.setDurationMs(CommonUtils.getCurrentMs() - startMs);
+    LOG.info("Bench threads finished");
+
+    service.shutdownNow();
+    service.awaitTermination(30, TimeUnit.SECONDS);
+
+    return mTotalResults;
+  }
+
+  private final class AlluxioNativeMaxFileThread implements Callable<Void> {
+    private static final int INIT_WAIT_TIME_MS = 100;
+    private static final int MAX_WAIT_TIME_MS = 15_000;
+    private static final int OPERATION_TIMEOUT_MS = 2_000;
+
+    private int mWaitTimeMs = INIT_WAIT_TIME_MS;
+    private final ExecutorService mExecutor = Executors.newSingleThreadExecutor();
+    private final MasterBenchTaskResult mResult = new MasterBenchTaskResult();
+
+    private final int mId;
+    private final FileSystem mFs;
+    private final Path mBasePath;
+    private final Path mFixedBasePath;
+
+    AlluxioNativeMaxFileThread(int id, FileSystem fs) {
+      mId = id;
+      mFs = fs;
+      mBasePath =
+          new Path(PathUtils.concatPath(mParameters.mBasePath, "files", mId));
+      mFixedBasePath = new Path(mBasePath, "fixed");
+      LOG.info("[{}]: basePath: {}, fixedBasePath: {}", mId, mBasePath, mFixedBasePath);
+    }
+
+    @Override
+    public Void call() throws Exception {
+      mResult.setRecordStartMs(CommonUtils.getCurrentMs());
+      AtomicLong localCounter = new AtomicLong();
+      while (!sFinish.get()) {
+        if (Thread.currentThread().isInterrupted()) {
+          break;
+        }
+        long increment = localCounter.getAndIncrement();
+        if (increment % 100_000 == 0) {
+          LOG.info("[{}] Created {} files", mId, increment);
+        }
+        try {
+          mExecutor.submit(() -> {
+            try {
+              applyOperation(increment);
```

Comment:
why not call `applyOperation()` directly to avoid threading cost?

Reply:
It does. If you want to timeout at a specific duration, try setting `alluxio.user.rpc.retry.max.duration`.

Reply:
The point is, `createFile` should not get stuck forever. Using a thread-pool to manage timeouts is redundant, while  `createFile` is guaranteed to be never get stuck unnecessarily. So I don't understand why do you need to have a timid loop there. 

2s operation time and 15s overall timeout also seems to be quite aggressive.

Reply:
Error handling depending on external thread-pool seems wrong, but if it doesn't work as such than I'm OK.